kitty/screen.c

```diff
@@ -236,6 +245,28 @@ screen_resize(Screen *self, unsigned int lines, unsigned int columns) {
     index_type x = self->cursor->x, y = self->cursor->y;
     LineBuf *n = realloc_lb(self->main_linebuf, lines, columns, &num_content_lines_before, &num_content_lines_after, self->historybuf, &x, &y, &self->as_ansi_buf);
     if (n == NULL) return false;
+
+
+    if (OPT(scrollback_fill_enlarged_window)) {
+      int lines_to_fill = (lines - self->main_linebuf->ynum) \
```

Comment:
Is that not handled by the next line?

```c
                          + (linebuf_continued_lines_count(self->main_linebuf) - linebuf_continued_lines_count(n));
```

Reply:
Depends on what you mean by work. That counts continued lines both above and below the cursor, so if thre are continued lines below it will change cursor position w.r.t the bottom of the screen.

Reply:
Would something like this address that issue?

```diff
diff --git a/kitty/line-buf.c b/kitty/line-buf.c
index 7ac76e9b..b0372436 100644
--- a/kitty/line-buf.c
+++ b/kitty/line-buf.c
@@ -327,9 +327,9 @@ is_continued(LineBuf *self, PyObject *val) {
 }
 
 unsigned int
-linebuf_continued_lines_count(LineBuf *self) {
+linebuf_continued_lines_count(LineBuf *self, index_type stop_at_line) {
     unsigned int count = 0;
-    for (unsigned int i = 0; i < self->ynum; i++)
+    for (unsigned int i = 0; i < self->ynum && i < stop_at_line; i++)
       if (self->line_attrs[i] & CONTINUED_MASK) count++;
 
     return count;
diff --git a/kitty/screen.c b/kitty/screen.c
index bb2a0b13..e94ce538 100644
--- a/kitty/screen.c
+++ b/kitty/screen.c
@@ -249,7 +249,7 @@ screen_resize(Screen *self, unsigned int lines, unsigned int columns) {
 
     if (OPT(scrollback_fill_enlarged_window)) {
       int lines_to_fill = (lines - self->main_linebuf->ynum) \
-                          + (linebuf_continued_lines_count(self->main_linebuf) - linebuf_continued_lines_count(n));
+                          + (linebuf_continued_lines_count(self->main_linebuf, y) - linebuf_continued_lines_count(n, y));
 
       while (lines_to_fill > 0) {
         if (self->historybuf->count <= 0) break;
```